src/boardlib/api/aurora.py

```diff
@@ -391,3 +361,224 @@ def save_climb(
+        else:
+            displayed_grade = None
+            difficulty = None
+        
+        final_logbook.append({
+            'board': bid_row['board'],
+            'angle': bid_row['angle'],
+            'climb_name': bid_row['climb_name'],
+            'date': bid_row['date'],
+            'logged_grade': None,
+            'displayed_grade': displayed_grade,
+            'difficulty': difficulty,
+            'tries': bid_row['tries'],
+            'is_mirror': bid_row['is_mirror'],
+            'is_ascent': False,
+            'comment': bid_row.get('comment', None)  # Use .get() to safely handle missing 'comment'
+        })
+    return final_logbook
+
+
+def calculate_sessions_count(group):
+    group = group.sort_values(by='date')
+    unique_dates = group['date'].dt.date.drop_duplicates().reset_index(drop=True)
+    sessions_count = unique_dates.rank(method='dense').astype(int)
+    sessions_count_map = dict(zip(unique_dates, sessions_count))
+    group['sessions_count'] = group['date'].dt.date.map(sessions_count_map)
+    return group
+
+
+def calculate_tries_total(group):
+    group = group.sort_values(by='date')
+    group['tries_total'] = group['tries'].cumsum()
+    return group
+
+
+def logbook_entries(board, username, password, grade_type="font", db_path=None):
+    login_info = login(board, username, password)
+    token = login_info["token"]
+    user_id = login_info["user_id"]
+    
+    bids_entries = list(bids_logbook_entries(board, username, password, db_path))
+    raw_ascents_entries = get_logbook(board, token, user_id)
+    
+    if not bids_entries and not raw_ascents_entries:
+        return pd.DataFrame(columns=['board', 'angle', 'climb_name', 'date', 'logged_grade', 'displayed_grade', 'difficulty', 'tries', 'is_mirror', 'is_ascent', 'comment'])
+
+    if bids_entries:
+        bids_df = pd.DataFrame(bids_entries)
+        bids_df['climbed_at'] = pd.to_datetime(bids_df['climbed_at'])
+        bids_summary = summarize_bids(bids_df, board)
+    else:
+        bids_summary = pd.DataFrame(columns=['climb_uuid', 'climb_name', 'date', 'is_mirror', 'angle', 'tries', 'board'])
+    
+    if raw_ascents_entries:
+        grades = get_grades(board)
+        grades_dict = {grade['difficulty']: grade for grade in grades}
+        ascents_entries = process_raw_ascent_entries(raw_ascents_entries, board, db_path, grades_dict, grade_type)
+        ascents_df = pd.DataFrame(ascents_entries)
+    else:
+        ascents_df = pd.DataFrame(columns=['board', 'angle', 'climb_uuid', 'name', 'date', 'logged_grade', 'difficulty', 'displayed_grade', 'tries', 'is_mirror', 'comment'])
+
+    final_logbook = combine_ascents_and_bids(ascents_df, bids_summary, db_path, grades_dict, grade_type)
+    
+    full_logbook_df = pd.DataFrame(final_logbook, columns=['board', 'angle', 'climb_name', 'date', 'logged_grade', 'displayed_grade', 'difficulty', 'tries', 'is_mirror', 'is_ascent', 'comment'])
```

Comment:
adding the climb_uuid would be pretty helpful here
This would simplify the attempts lookup for climbdex a lot: https://github.com/lemeryfertitta/Climbdex/issues/55

`full_logbook_df = pd.DataFrame(final_logbook, columns=['climb_uuid', 'board', 'angle', 'climb_name', 'date', 'logged_grade', 'displayed_grade', 'difficulty', 'tries', 'is_mirror', 'is_ascent', 'comment'])`

Reply:
I think that is a fine suggestion to make this more useful for Climbdex, but I would want to make sure we don't do that in a way that clutters the output of the boardlib `logbook` command. Realistically, users of the command line version don't need the climb UUID in the output CSV.

One idea would be to refactor out a version of this function that does everything you've suggested here, and then in the command-line version we can handle the extra steps. From my understanding, that would just be to login and get a token and also to filter out the climb_uuid. I'd be happy to accept a PR for this if it would help with your proposed climbdex additions!

Reply:
Alright, will do it this way!